presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/ScaledWriterRule.java

```diff
@@ -40,10 +41,23 @@ public boolean isEnabled(Session session)
         return useHBOForScaledWriters(session);
     }
 
+    @Override
+    public boolean isCostBased(Session session)
+    {
+        return true;
+    }
+
+    @Override
+    public String getStatsSource()
+    {
+        return statsSource;
+    }
+
     @Override
     public Result apply(TableWriterNode node, Captures captures, Context context)
     {
         double taskNumber = context.getStatsProvider().getStats(node).getTableWriterNodeStatsEstimate().getTaskCountIfScaledWriter();
+        statsSource = context.getStatsProvider().getStats(node).getSourceInfo().getSourceInfoName();
```

Comment:
since this construct is used several times I wonder if we should introduce a utility function in PlannerUtils:getStatsSource(Context context, PlanNode node)